packages/orchestration/test/test-chain-name-hub.js

```diff
@@ -0,0 +1,213 @@
+  agoriclocal: 'agoric',
+};
+
+const makeChainNames = async () => {
+  // XXX get a mock agoricNames.brands,
+  const { nameHub: chainNames, nameAdmin: chainNamesAdmin } = makeNameHubKit();
+  for (const [allegedChainName, value] of Object.entries(mockChainInfo)) {
+    const { bondDenom, chainId, denoms, connections } = value;
+    const child = await chainNamesAdmin.provideChild(allegedChainName, [
+      'chainId',
+      'bondAsset',
+      'brand',
+      'denom',
+      'trace',
+      'connection',
+    ]);
+    child.nameAdmin.update('chainId', chainId);
+    child.nameAdmin.update(
+      'bondAsset',
+      harden({
+        brand: 'XXX_agoricNames_brand', // FIXME, use ERTP brand
+        denom: bondDenom,
+      }),
+    );
+    const { nameHub: brandNames, nameAdmin: brandNamesAdmin } =
+      makeNameHubKit();
+    const { nameHub: denomNames, nameAdmin: denomNamesAdmin } =
+      makeNameHubKit();
+    const { nameHub: traceNames, nameAdmin: traceNamesAdmin } =
+      makeNameHubKit();
+
+    for (const { denom, path, baseDenom } of denoms) {
+      const brandName = denomsToBrandName[baseDenom || denom];
+      await brandNamesAdmin.provideChild(denom);
+      brandNamesAdmin.update(denom, brandName); // FIXME, use ERTP Brand
+
+      await denomNamesAdmin.provideChild(brandName);
+      denomNamesAdmin.update(brandName, denom);
+
+      await traceNamesAdmin.provideChild(denom);
+      traceNamesAdmin.update(denom, harden({ baseDenom, path }));
+      await traceNamesAdmin.provideChild(brandName);
+      traceNamesAdmin.update(brandName, harden({ baseDenom, path }));
+    }
+    child.nameAdmin.update('brand', brandNames);
+    child.nameAdmin.update('denom', denomNames);
+    child.nameAdmin.update('trace', traceNames);
+
+    const { nameHub: connectionNames, nameAdmin: connectionNamesAdmin } =
+      makeNameHubKit();
+    for (const {
+      connectionInfo,
+      transferChannels,
+      chainId: lChainId,
+      lastUpdated,
+    } of connections) {
+      const chainName = chainIdToChainName[lChainId];
+      await connectionNamesAdmin.provideChild(chainName);
+      connectionNamesAdmin.update(
+        chainName,
+        harden({
+          chainId: lChainId,
+          connectionInfo: {
+            ...connectionInfo,
+            ...lastUpdated,
+          },
+          // FIXME, there can be more than one, but typically only one canonical
+          transferChannel: transferChannels[0],
+        }),
+      );
+    }
+    child.nameAdmin.update('connection', connectionNames);
+  }
+  return { chainNames, chainNamesAdmin };
+};
+
+test('chain-name-hub', async t => {
+  const { chainNames } = await makeChainNames();
+
+  const chainId = await chainNames.lookup('osmosis', 'chainId');
+  t.is(chainId, 'osmosis-test');
+  const bondAsset = await chainNames.lookup('osmosis', 'bondAsset');
+  t.deepEqual(bondAsset, { brand: 'XXX_agoricNames_brand', denom: 'uosmo' });
+
+  // lookup up Brand from denom
```

Comment:
That's what vbankAsset is for, no?

To look up `uosmo`, you'd have to know the channel to compute the relevant ibc denom first, of course.